test/e2e/cache_watch_test.go

```diff
@@ -21,32 +22,38 @@ import (
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/data"
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/k8s"
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/model"
-	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/e2e/utils"
 )
 
+type tc struct {
+	namespaces      []string
+	watchNamespaces []string
+	wantToFind      []string
+}
+
 var _ = Describe("Kubernetes cache watch test:", Label("cache-watch"), func() {
 	DescribeTable("Cache gets", Label("watch-gets"),
-		func(name string, namespaceIndexesToWatch []int, expectedNamespaceFound []bool) {
+		func(ctx context.Context, testCase *tc) {
 			testData := model.DataProvider(
-				fmt.Sprintf("cache-%s", name),
+				fmt.Sprintf("cache-%s", CurrentSpecReport().LeafNodeLabels[0]),
```

Comment:
in case you're wondering what this returns: `CurrentSpecReport().LeafNodeLabels[0]` returns `"watch-gets"` in this case which is the leaf label used for this table driven test.